debugger/src/debug_adapter/repl_commands.rs

```diff
@@ -0,0 +1,774 @@
+        if let Some(format_specifier) = format_specifier {
+            result.format_specifier = format_specifier;
+        }
+        if let Some(unit_specifier) = unit_specifier {
+            result.unit_specifier = unit_specifier;
+        }
+        if !nuf.is_empty() {
+            result.unit_count = nuf.parse::<usize>().map_err(|error| {
+                DebuggerError::ReplError(format!("Invalid unit count specifier: {value} - {error}"))
+            })?;
+        }
+
+        Ok(result)
+    }
+}
+
+struct GdbNufMemoryResult<'a> {
+    nuf: &'a GdbNuf,
+    memory: &'a Vec<u8>,
+}
+
+impl Display for GdbNufMemoryResult<'_> {
+    // TODO: Consider wrapping lines at 80 characters.
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self.nuf.format_specifier {
+            GdbFormat::Binary => {
+                let width = 10_usize;
+                for byte in self.memory {
+                    write!(f, "{byte:#0width$b} ")?;
+                }
+            }
+            GdbFormat::Hex => {
+                let width = 4_usize;
+                for byte in self.memory {
+                    write!(f, "{byte:#0width$x} ")?;
+                }
+            }
+            _ => write!(f, "<cannot print>")?,
+        }
+        Ok(())
+    }
+}
+
+static REPL_COMMANDS: &[ReplCommand<H>] = &[
+    ReplCommand {
+        command: "help",
+        help_text: "Information about available commands and how to use them.",
+        sub_commands: None,
+        args: None,
+        handler: |_, response_body, _, _| {
+            let mut help_text =
+                "Usage:\t-Use <Ctrl+Space> to get a list of available commands.".to_string();
+            help_text.push_str("\n\t- Use <Up/DownArrows> to navigate through the command list.");
+            help_text.push_str("\n\t- Use <Hab> to insert the currently selected command.");
+            help_text.push_str("\n\t- Note: This implementation is a subset of gdb commands, and is intended to behave similarly.");
+            help_text.push_str("\nAvailable commands:");
+            for command in REPL_COMMANDS {
+                help_text.push_str(&format!("\n{command}"));
+            }
+            response_body.result = help_text;
+            Ok(DebugSessionStatus::Continue)
+        },
+    },
+    ReplCommand {
+        command: "quit",
+        help_text: "Disconnect (and suspend) the target.",
+        sub_commands: None,
+        args: None,
+        handler: |target_core, response_body, _, _| {
+            target_core.core.halt(Duration::from_millis(500))?;
+            response_body.result = "Debug Session Terminated".to_string();
+            Ok(DebugSessionStatus::Terminate)
+        },
+    },
+    ReplCommand {
+        command: "c",
+        help_text: "Continue running the program on the target.",
+        sub_commands: None,
+        args: None,
+        handler: |target_core, response_body, _, _| {
+            target_core.core.run()?;
```

Comment:
Is this synchronized with the VS Code debugger state?

Reply:
Yes. The debugger constantly polls for state, and based on change of state, it will send the appropriate event to synch with VSCode. So if you use this 'c' command, it will resume the core, and then on the next poll, VSCode will be notified of the running status, or if a new breakpoint was hit before then, will update VSCode with the new halt reason, and so on.